--- a/gsm/src/fcst/wrisig.F
+++ b/gsm/src/fcst/wrisig.F
@@ -145,6 +145,9 @@
       dimension ak5x(levp1_),bk5x(levp1_)
 c
       dimension orog(lnt22_)
+      integer iorog_read
+      save orog, iorog_read
+      data iorog_read/0/
 c
       parameter(kdum=201-levs_-1-levs_,kdum2=21,kens=2)
       dimension dummy(kdum),dummy2(kdum2),ensemble(kens)
@@ -156,6 +159,9 @@
       data fni/'sig','sigp','sigi','sigit','sigitdt'/
       data nchi/3,4,4,5,7/
       character*128 fno
+      integer npes,ncol,nrow,mype
+      common /commpi/ npes,ncol,nrow,mype
+      if (mype .ne. 0) return
       if (itpdt .eq. 1 .or. itpdt .eq. 4 ) then
         call sph_fft_driver(1,slq_q1,rq,ntotal_)
       else if (itpdt.eq.2 .or. itpdt.eq.5) then
@@ -164,6 +170,7 @@
 c
 c  get topography from history file
 c
+      if (iorog_read .eq. 0) then
       close(n)
       rewind n
       open(unit=n,file='./sigit',form='unformatted',err=700)
@@ -180,6 +187,8 @@
       read(n)( orog(i),i=1,lnt2_)
       print *,' read gz '
       close(n)
+      iorog_read = 1
+      endif
 c
       if(itpdt.lt.4) then
         call fnam(fni(itpdt),nchi(itpdt),fhour,fno,ncho)
--- a/gsm/src/fcst_par/Makefile.in
+++ b/gsm/src/fcst_par/Makefile.in
@@ -12,7 +12,7 @@
 CHMOD = @CHMOD@

 CC = @CC@
-CPP = @CPP@
+CPP = @CPP@ -DMP
 F77 = @MODEL_MPI_F77@
 FORT_FLAGS = @MODEL_MPI_FORT_FLAGS@
 LOAD_FLAGS = @MODEL_MPI_LOAD_FLAGS@
