--- a/gsm/src/fcst/wrisig.F
+++ b/gsm/src/fcst/wrisig.F
@@ -104,4 +104,7 @@
       dimension ak5x(levp1_),bk5x(levp1_)
 c
       dimension orog(lnt22_)
+      integer iorog_read
+      save orog, iorog_read
+      data iorog_read/0/
 c
@@ -135,7 +138,8 @@
 c
 c  get topography from history file
 c
 #ifdef MP
       if( mype.eq.master ) then
 #endif
+      if (iorog_read .eq. 0) then
       close(n)
@@ -168,6 +172,8 @@
       print *,' read gz '
 #endif
       close(n)
+      iorog_read = 1
+      endif
 #ifdef MP
       endif
 #endif
--- a/gsm/src/fcst_par/Makefile.in
+++ b/gsm/src/fcst_par/Makefile.in
@@ -12,7 +12,7 @@
 CHMOD = @CHMOD@

 CC = @CC@
-CPP = @CPP@
+CPP = @CPP@ -DMP
 F77 = @MODEL_MPI_F77@
 FORT_FLAGS = @MODEL_MPI_FORT_FLAGS@
 LOAD_FLAGS = @MODEL_MPI_LOAD_FLAGS@
