--- a/gsm/src/fcst/wrisig.F
+++ b/gsm/src/fcst/wrisig.F
@@ -119,6 +119,11 @@
       data fni/'sig','sigp','sigi','sigit','sigitdt'/
       data nchi/3,4,4,5,7/
       character*128 fno
+#ifdef MP
+      integer npes,ncol,nrow,mype
+      common /commpi/ npes,ncol,nrow,mype
+      if (mype .ne. 0) return
+#endif

 #ifdef NISLQ
       if (itpdt .eq. 1 .or. itpdt .eq. 4 ) then
--- a/gsm/src/fcst_par/Makefile.in
+++ b/gsm/src/fcst_par/Makefile.in
@@ -12,7 +12,7 @@
 CHMOD = @CHMOD@
 
 CC = @CC@
-CPP = @CPP@
+CPP = @CPP@ -DMP
 F77 = @MODEL_MPI_F77@
 FORT_FLAGS = @MODEL_MPI_FORT_FLAGS@
 LOAD_FLAGS = @MODEL_MPI_LOAD_FLAGS@
